lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
+    attempts = 10
+    loop do
+      result = get_attributes(path,false)
+      # return false for file not found
+      return false if result == INVALID_FILE_ATTRIBUTES
+      # return true if we found a file/dir and it's not a symlink
```

Comment:
I think your logic is right, though someone reading it might think it a little peculiar.  Might want to leave a note about all the other file attributes that could be set:  https://msdn.microsoft.com/en-us/library/windows/desktop/gg258117(v=vs.85).aspx